src/connectors/kessel/impl.js

```diff
@@ -0,0 +1,200 @@
+                builder.unauthenticated();
+            }
+
+            // Build the Client
+            this.kesselClient = builder.buildAsync();
+            this.initialized = true
+            log.info('Kessel client initialized successfully using ClientBuilder');
+
+            return this.kesselClient
+        } catch (error) {
+            log.error({ error }, 'Failed to initialize Kessel client');
+            this.initialized = false;
+        }
+    }
+
+    async checkSinglePermission(userId, workspaceId, relation) {
+        try {
+            // Create the check request using the new API structure
+            const checkRequest = {
+                subject: {
+                    resource: {
+                        reporter: {
+                            type: "rbac"
+                        },
+                        resourceId: `${this.kesselConfig.principalDomain}/${userId}`,
+                        resourceType: "principal"
+                    }
+                },
+                object: {
+                    reporter: {
+                        type: "rbac"
+                    },
+                    resourceId: workspaceId,
+                    resourceType: "workspace"
+                },
+                relation: relation
+            };
+
+            // Use async/await with the new client
+            const response = await this.kesselClient.check(checkRequest);
+            return response.allowed === Allowed.ALLOWED_TRUE;
+        } catch (error) {
+            log.warn({ error, userId, workspaceId, relation }, 'Kessel check call failed');
+            throw error;
+        }
+    }
+
+
+    getIdentityFromHeaders() {
+        const headers = this.getForwardedHeaders();
+
+        // Try to extract identity from x-rh-identity header
+        if (headers['x-rh-identity']) {
+            try {
+                const identityHeader = Buffer.from(headers['x-rh-identity'], 'base64').toString();
+                return JSON.parse(identityHeader);
+            } catch (error) {
+                log.warn({ error }, 'Failed to parse x-rh-identity header');
+            }
+        }
+
+        return null;
+    }
+
+    // Compatibility method to check specific permission
+    async hasPermission(resource, action, subject_org_id) {
+        if (!this.kesselConfig.enabled || !this.initialized || !this.kesselClient) {
+            return false;
+        }
+
+        const startTime = Date.now();
+        try {
+            // Convert traditional RBAC permission to workspace permission
+            const workspacePermission = this.convertRbacToWorkspacePermission(resource, action);
+            if (!workspacePermission) {
+                log.warn({ resource, action }, 'Unknown permission mapping');
+                return false;
+            }
+
+            // Get workspace ID from subject_org_id or use default
+            const workspace = await this.getDefaultWorkspaceIdForSubject(subject_org_id);
```

Comment:
💯 

This `org_id` usage is correct, you call `rbac` with the `org_id` to fetch the default `workspace id`